src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -385,7 +394,7 @@ public override void DidReceiveResponse (NSUrlSession session, NSUrlSessionDataT
 						inflight.Stream.TrySetException (new ObjectDisposedException ("The content stream was disposed."));
 
 						sessionHandler.RemoveInflightData (dataTask);
-					}, inflight.CancellationToken);
+					}, inflight.CancellationTokenSource.Token);
```

Comment:
Real interesting part, do no use the now from the SendAsync, use or new one. The two are connected, see above comment.